iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/interpstring/InterpolatedStringMiddlePiece.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep.interpstring;
+
+import com.sonar.sslr.api.typed.Optional;
+import org.sonar.iac.arm.tree.api.Expression;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.common.api.tree.Tree;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class InterpolatedStringMiddlePiece {
+  private final Expression expression;
+  private final SyntaxToken rCurly;
+  @Nullable
+  private final SyntaxToken stringChars;
+  private final SyntaxToken dollarLcurly;
+
+  public InterpolatedStringMiddlePiece(Expression expression, SyntaxToken rCurly, Optional<SyntaxToken> stringChars, SyntaxToken dollarLcurly) {
+    this.expression = expression;
+    this.rCurly = rCurly;
```

Comment:
IMO value or text will a better name for it. Please also add accessor / getter for it.

Reply:
`StringMiddlePiece` contains multiple pieces of text and also expressions. I think it seems not very useful to have an accessor for text instead of the whole `interpString` contents; but since `Expression` doesn't extend `TextTree`, it's not obvious how to construct an accessor for the whole content.